### PR TITLE
Added SonarScanner which wraps sonar-scanner-api

### PR DESCRIPTION
**Pull request by DoNotWorr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,19 +2,20 @@
...
-> token=botToken123
+### Discord Bot
 
-## Sonarqube server
+> token=botToken123
 
-***application.properties*** file inside the resources folder:
+### Sonarqube server
 
-> sonarqubeAPI=port/api<br>
```
Comment (DoNotWorr) [inline README.md:18]: I separated public and secret application properties.
This property is in the public file and I don't think we need to explain those.

File: pom.xml [hunk 2]
```diff
@@ -1,115 +1,120 @@
...
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-hateoas</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.sonarsource.scanner.api</groupId>
+            <artifactId>sonar-scanner-api</artifactId>
+            <version>2.16.2.588</version>
+        </dependency>
+    </dependencies>
```
Comment (DoNotWorr) [inline pom.xml:66]: Added this.

File: src/main/java/org/five/sonarqubot/controllers/HttpRequests.java [hunk 3]
```diff
@@ -12,14 +11,12 @@
 
 
 @RestController
-@PropertySource("classpath:app.properties")
```
Comment (DoNotWorr) [inline src/main/java/org/five/sonarqubot/controllers/HttpRequests.java:15]: Not needed for this file.

File: src/main/java/org/five/sonarqubot/events/FileService.java [hunk 4]
```diff
@@ -7,6 +7,5 @@ public interface FileService {
 
     void createDirectory();
 
-
-    Mono<String> createFile(String code, String filename);
+    Mono<String> createFile(String code);
```
Comment (DoNotWorr) [inline src/main/java/org/five/sonarqubot/events/FileService.java:10]: File name is now generated inside the method.

File: src/main/java/org/five/sonarqubot/events/MessageService.java [hunk 5]
```diff
@@ -4,7 +4,7 @@
 import reactor.core.publisher.Mono;
 
 public interface MessageService {
-    public Mono<Message> onlyCodeMessages(Message eventMessage);
+    Mono<Message> onlyCodeMessages(Message eventMessage);
```
Comment (DoNotWorr) [inline src/main/java/org/five/sonarqubot/events/MessageService.java:7]: Not needed for interfaces

**Review by DoNotWorr — COMMENTED** [review]
Added some comments to provide context for code review.
Resolved the conversations, but they can be viewed when doing review.

**Review by Abdih10 — APPROVED** [review]
It looks good to me!

**Review by IsaLjungdahl — APPROVED** [review]
Looking great! Well done 🏅

**Pull request merged**